core/src/main/scala/kafka/utils/ShutdownableThread.scala

```diff
@@ -63,6 +65,8 @@ abstract class ShutdownableThread(val name: String, val isInterruptible: Boolean
         doWork()
       }
     } catch{
+      case fatalExitError: FatalExitError =>
+        fatalExitError.shutdownSystem()
```

Comment:
Shutdown hook waits on kafka.utils.ShutdownableThread.awaitShutdown(ShutdownableThread.scala:52)
I've attached stack trace to https://issues.apache.org/jira/browse/KAFKA-4039